examples/stable_diffusion_v2/train_text_to_image.py

```diff
@@ -140,22 +127,164 @@ def load_pretrained_model_vae_unet_cnclip(pretrained_ckpt, cnclip_ckpt, net):
         logger.warning(f"Checkpoint file {pretrained_ckpt}, {cnclip_ckpt} dose not exist!!!")
 
 
+def _check_cfgs_in_parser(cfgs: dict, parser: argparse.ArgumentParser):
+    actions_dest = [action.dest for action in parser._actions]
+    defaults_key = parser._defaults.keys()
+    for k in cfgs.keys():
+        if k not in actions_dest and k not in defaults_key:
+            raise KeyError(f"{k} does not exist in ArgumentParser!")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--train_config",
+        default="",
+        type=str,
+        help="train config path to load a yaml file that override the default arguments",
+    )
+    parser.add_argument(
+        "-v",
+        "--version",
```

Comment:
How is --version affect the pretrained model weight selection here? I think we should either delete it or use it to select the pretrained weight version to finetune from